GiEnJul/Infrastructure/AutoMapperProfile.cs

```diff
@@ -31,9 +31,8 @@ public AutoMapperProfile()
                 .ForMember(dest => dest.MatchedGiver, act => act.Ignore());
 
             CreateMap<Models.Recipient, Entities.Recipient>()
-                .ConstructUsing(x => new Entities.Recipient(x.Location, x.EventName))
-                .ForMember(dest => dest.RowKey, act => act.Ignore())
-                .ForMember(dest => dest.PartitionKey, act => act.Ignore())
+                .ForMember(dest => dest.RowKey, opt => opt.Condition(src => (!string.IsNullOrEmpty(src.RowKey))))
+                .ForMember(dest => dest.PartitionKey, opt => opt.Condition(src => (!string.IsNullOrEmpty(src.PartitionKey))))
```

Comment:
I have never used opt.Condition() before, what happens if one of these values are empty?

Reply:
It was added to prevent the mapping from creating a new entity when an entity is updated. So if the field is empty, then a new rowKey and partitionKey is created, if not it maps. Makes sense?